docs/program/apis/data-client.md

```diff
@@ -0,0 +1,246 @@
+---
+title: "Upload and Retrieve Data with Viam's Data Client API"
+linkTitle: "Data Client"
+weight: 10
+type: "docs"
+description: "Use the data client API to upload and retrieve data directly to the Viam app."
+tags:
+  [
+    "cloud",
+    "sdk",
+    "viam-server",
+    "networking",
+    "apis",
+    "robot api",
+    "data management",
+    "data",
+  ]
+---
+
+The data client API allows you to upload and retrieve data to and from the [Viam app](https://app.viam.com).
+
+{{% alert title="Support Notice" color="note" %}}
+
+Data client API methods are only available in the Python SDK.
+
+{{% /alert %}}
+
+## Establish a connection
+
+To use the Viam data client API, you first need to instantiate a [`ViamClient`](https://python.viam.dev/autoapi/viam/app/viam_client/index.html#viam.app.viam_client.ViamClient) and then instantiate an [`DataClient`](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient).
+See the following example for reference.
+To find the location secret, go to [Viam app](https://app.viam.com/), and go to the [**Code sample**](https://docs.viam.com/manage/fleet/robots/#code-sample) tab of any of the robots in the location.
+Toggle **Include secret** on and copy the `payload`.
+For the URL, use the address of any of the robots in the location (also found on the **Code sample** tab).
+
+```python {class="line-numbers linkable-line-numbers"}
+import asyncio
+
+from viam.rpc.dial import DialOptions, Credentials
+from viam.app.viam_client import ViamClient
+
+
+async def connect() -> ViamClient:
+    dial_options = DialOptions(
+        # The URL of any robot in the location.
+        auth_entity='beepboop-main.YOUR LOCATION ID.viam.cloud',
+        credentials=Credentials(
+            type='robot-location-secret',
+            # The location secret
+            payload='YOUR LOCATION SECRET'
+        )
+    )
+    return await ViamClient.create_from_dial_options(dial_options)
+
+
+async def main():
+
+    # Make a ViamClient
+    viam_client = await connect()
+    # Instantiate a DataClient called "data" to run data client API methods on
+    data_client = viam_client.data_client
+
+    viam_client.close()
+
+if __name__ == '__main__':
+    asyncio.run(main())
+```
+
+Once you have instantiated a `DataClient`, you can run the following [API methods](#api) against the `DataClient` object (named `data_client` in the examples).
```

Comment:
opt: or, since we explicitly refer to the client by name here, the code comment above could instead mirror:

> Instantiate a DataClient called "data_client" to run data client API methods on

Very opt, just like the consistency.